src/components/elements/inputs/input.js

```diff
@@ -27,28 +21,36 @@ export default class Input extends Component {
     }
   }
 
-  getValue(target) {
-    return target.value
-  }
+  render() {
+    const Element = factory(this.props.property)
+    let className = 'ct-input'
 
-  onBlur() {
-    this.update(this.state.value)
-  }
+    if (this.props.property.errors.length) {
+      className += ' ct-input-invalid'
+    }
 
-  onChange(e) {
-    this.setState({
-      value: e.target.value === '' ? null : this.getValue(e.target),
-    })
+    return (
+      <Element
+        className={className}
+        property={this.props.property}
+        save={value => this.save(value)}
+        update={value => this.setState({ value })}
+        value={this.state.value} />
+    )
   }
 
-  update(value) {
-    this.setState({ value })
+  save(value) {
+    if (value === undefined) {
+      value = this.state.value
+    }
+    else {
+      this.setState({ value })
+    }
 
-    if (this.props.property.value != value) {
-      const property = this.props.property
-      const updateHref = property.links.update.href
+    const { property } = this.props
 
-      this.props.update(updateHref, property.id, value)
+    if (property.value !== value) {
+      this.props.update(property.links.update.href, property.id, value)
```

Comment:
Now when I think about it, it's a little confusing that the `save` function calls `update` in the api but the `update` function doesn't. I know `update` means different things in these places but it's still a bit confusing because it's the same word..

Maybe we should reconsider the `update` and `save` functions in the components again.. I know I changed my mind since yesterday but now I'm leaning towards calling them `onChange` and `onCommit` or something. The `on` prefix would make a lot of sense when you create the element in this file.

I like this:
```javascript
return (
   <Element
     className={className}
     property={this.props.property}
     onCommit={value => this.save(value)}
     onChange={value => this.setState({ value })}
     value={this.state.value} />
)
```